src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx

```diff
@@ -2126,8 +2138,12 @@ function processInputData(
   binSpec?: BinSpec,
   binWidthSlider?: BinWidthSlider,
   overlayVariable?: Variable,
-  colorPaletteOverride?: string[]
+  colorPaletteOverride?: string[],
+  dependentIsProportion?: boolean
```

Comment:
An alternative to passing yet another argument might be to create at io-ts type for the numeratorN/denominatorN structure, so:

```
      binSampleSize: union([
        array(
          type({
            N: number,
          })
        ),
        array(
          type({
            numeratorN: number,
            denominatorN: number,
          })
        ),
      ]),
```
would become
```
export type ProportionSampleSizes = TypeOf<typeof ProportionSampleSizes>;
const ProportionSampleSizes = type({
            numeratorN: number,
            denominatorN: number,
          });

...

      binSampleSize: union([
        array(
          type({
            N: number,
          })
        ),
        array(ProportionSampleSizes),
      ]),
```

Then in the `markerSymbol` function you could replace the `dependentIsProportion` with something like

```
el.binSampleSize.every((bss) => ProportionSampleSizes.is(bss))
```

I don't know if that makes it any more (or less) readable... (and it probably doesn't fix all the TS issues)

It would be nice to avoid the `any` and the `hasOwnProperty` check on just the first array element. But I may just be introducing more TS spaghetti to compensate...

Reply:
I might wrap back around to the `numeratorN` stuff, though there are more pressing TS issues, like the many `any`s, as you mention.

Reply:
I think we might need to explicitly "tag" the LineplotResponse series with an "isZeroSeries" property?

Introspecting to figure out which symbol to use is tricky. I'm noticing a `el.binSampleSize[0] is undefined` error when the zeroSeries is empty, though that may be avoidable with a length check - I haven't tried that.

Here's how to reproduce:
LLINEUP -> x=observation date, y=Plasmodium by RDT, num=Yes, denom=all
(basically there are no 0/0s)